packages/base/lua/Profiler.lua

```diff
@@ -0,0 +1,310 @@
+			customWarning(string.format("Block '%s' has already been started. Please, stop the block before re-start it.", block.name))
+			stop(block.name)
+		elseif not block then
+			block = createBlock(name)
+			self.blocks[name] = block
+		end
+
+		block:start()
+		Stack.push(self.stack, block)
+	end,
+	--- Return the current block.
+	-- @usage Profiler():start("block")
+	-- print(Profiler():current().name) -- block
+	-- Profiler():stop("block")
+	current = function(self)
+		return Stack.top(self.stack)
+	end,
+	--- Return how many times a given block has started.
+	-- @arg name A string with the block name. If the name is not informed, then it returns the count of the current block.
+	-- @usage Profiler():start("block")
+	-- print(Profiler():count("block")) -- 1
+	-- Profiler():stop("block")
+	count = function(self, name)
+		optionalArgument("name", "string", name)
+		local block = self.blocks[name or self:current().name]
+		if not block then
+			customError(string.format("Block '%s' was not found.", name))
+		end
+
+		return block.count
+	end,
+	--- Return how much time was spent on the block.
+	-- It returns two representations: a string with a human-like representation of the time
+	-- and a number with the time in seconds.
+	-- @arg name A string with the block name. If the name is not informed, then it returns the uptime of the current block.
+	-- @usage Profiler():start("block")
+	-- Profiler():stop("block")
+	-- stringTime, numberTime = Profiler():uptime("block")
+	uptime = function(self, name)
+		optionalArgument("name", "string", name)
+		local block = self.blocks[name or self:current().name]
+		if not block then
+			customError(string.format("Block '%s' was not found.", name))
+		end
+
+		local time = block:uptime()
+		return timeToString(time), time
+	end,
+	--- Stop to measure the time of a given block. It also returns how much time was spent with the block
+	-- in two representations: a string with a human-like representation of the time and a number with the time in seconds.
+	-- @arg name A string with the block name. If the name is not informed, then it stops and return the uptime of the current block.
+	-- @usage Profiler():start("block")
+	-- stringTime, numberTime = Profiler():stop("block")
+	stop = function(self, name)
+		optionalArgument("name", "string", name)
+		if name == "main" or (not name and self:current() and self:current().name == "main") then
+			customWarning("The block 'main' cannot be stopped.")
+			return self:uptime(name)
+		end
+
+		local block
+		if not name then
+			block = Stack.pop(self.stack)
+		elseif self.blocks[name] then
+			block = self.blocks[name]
+			Stack.remove(self.stack, block)
+		else
+			customError(string.format("Block '%s' was not found.", name))
+		end
+
+		if block.running then
+			block:stop()
+		end
+
+		local time = block:uptime()
+		return timeToString(time), time
+	end,
+	--- Clean the Profiler, removing all blocks and restarting its execution time.
+	-- @usage Profiler():clean()
```

Comment:
Probably this clean here is affecting the rest of the tests, because it executes in the singleton. Use `-- DONTRUN` in the first line of usage to avoid running it.